### ISSUE TITLE
Static variables are not shown correctly in code hints

### ISSUE THREAD

**Issue by ratkingsminion** [godotengine/godot#85046]
### Godot version

v4.1.3.stable.official [f06b6836a]

### System information

Windows 10

### Issue description

When trying to get a code hint in Godot's code editor for a static variable in a script, the variable isn't shown in the list. It works for functions. When creating an instance of the script via .new(), then the static variable is shown (also alongside with the function, which is also wrong).

### Steps to reproduce

Write a test script like this:

```gdscript
class_name Test

static var variable:int = 1

static func function():
	print(variable)
```

In another script, try to get code hint by writing `Test.` - it won't show variable.

![Screenshot_8](https://github.com/godotengine/godot/assets/3142253/a2251772-07e3-4cbe-b053-d9a542ae2917)

Writing `Test.new().` will show variable. And also function, which it shouldn't either.

![Screenshot_9](https://github.com/godotengine/godot/assets/3142253/277828c9-293f-462a-b4f8-0403f95d075a)

### Minimal reproduction project

N/A

**Comment by dalexeev** [comment]
* Please check it in 4.2 rc 1. It was fixed by #83150, if I remember correctly.

**Comment by ratkingsminion** [comment]
Yes, looks like it. Thanks!
